lib/Convos/Core/Backend/File.pm

```diff
@@ -80,13 +80,13 @@ sub messages {
   $re = qr{\Q$re\E}i unless ref $re;
   $re = qr/^(\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}) (.*$re.*)$/;
 
-  $args{after}  = Time::Piece->strptime($query->{after},  '%Y-%m-%dT%H:%M:%S')if $query->{after};
-  $args{before} = Time::Piece->strptime($query->{before}, '%Y-%m-%dT%H:%M:%S') if $query->{before};
-  # If neither before nor after are defined, search everything before now
-  $args{before} = gmtime if !$args{before} && !$args{after};
   $args{limit}    = $query->{limit} || 60;
   $args{messages} = [];
   $args{re}       = $re;
+  # TODO: What to do when after is after before?
+  $args{before}   = $query->{before} ? Time::Piece->strptime($query->{before}, '%Y-%m-%dT%H:%M:%S') : gmtime;
+  $args{after}    = $query->{after}  ? Time::Piece->strptime($query->{after},  '%Y-%m-%dT%H:%M:%S') : $args{before}->add_months(-1);
```

Comment:
So yeah, I'm not sure what behavior you're wanting, so please let me know that.

To explain the code (since it wasn't commented...  :(

If the query does not provide "before" default to now (that was original behavior).
If the query does not provide "after" default to one month before "before" (meaning a 1-2 month time span).

My reason for implementing this strict behavior (to enforce the presence of "before" *and* "after") was to prevent deep recursion, to go from now until the beginning of time.  Instead, this puts a hard stop-gap on the recursion, but also on length of search (just be explicit about the before / after).  The other benefit to always searching between "after" and "before" is that we can focus on just one type of file reading: File::ReadBackwards